packages/components/data-table/src/cell.tsx

```diff
@@ -1,18 +1,53 @@
-import PropTypes from 'prop-types';
-import requiredIf from 'react-required-if';
+import { ReactNode } from 'react';
 import {
   RightTriangleFilledIcon,
   RightTriangleLinearIcon,
 } from '@commercetools-uikit/icons';
-import { filterDataAttributes } from '@commercetools-uikit/utils';
+import { filterDataAttributes, warning } from '@commercetools-uikit/utils';
 import { BaseCell, CellInner, RowExpandCollapseButton } from './cell.styles';
 
 import Resizer from './column-resizer';
 
-const DataCell = (props) => {
+export type TDataCell = {
+  children: ReactNode;
+  isCondensed?: boolean;
+  isTruncated?: boolean;
+  onCellClick?: () => void;
+  shouldIgnoreRowClick?: boolean;
+  verticalCellAlignment?: 'top' | 'center' | 'bottom';
+  horizontalCellAlignment?: 'left' | 'center' | 'right';
+  shouldRenderBottomBorder: boolean;
+  shouldRenderCollapseButton: boolean;
+  shouldRenderResizingIndicator: boolean;
+  handleRowCollapseClick?: () => void;
+  isRowCollapsed?: boolean;
+};
+
+type TDefaultProps = {
+  isTruncated: boolean;
+  shouldRenderBottomBorder: boolean;
+};
+const defaultProps: TDefaultProps = {
+  isTruncated: false,
+  shouldRenderBottomBorder: true,
+};
+
+const DataCell = (props: TDataCell) => {
+  if (props.shouldRenderCollapseButton) {
+    warning(
+      typeof props.handleRowCollapseClick === 'function',
+      'DataTable: "handleRowCollapseClick" is required when shouldRenderCollapseButton is true.'
+    );
+    warning(
+      props.isRowCollapsed,
```

Comment:
```suggestion
      typeof props.isRowCollapsed === 'boolean',
```

otherwise it won't work when passing `false`.